Tasks/VsTest/settingshelper.ts

```diff
@@ -187,23 +194,42 @@ export async function updateSettingsFileAsRequired(settingsFile: string, isParal
     return defer.promise;
 }
 
-function updateRunSettingsWithParameters(result: any, overrideParametersString: string) {
+function updateSettingsWithParameters(result: any, overrideParametersString: string) {
     const overrideParameters = parameterParser.parse(overrideParametersString);
-    if (result.RunSettings && result.RunSettings.TestRunParameters && result.RunSettings.TestRunParameters[0] &&
-        result.RunSettings.TestRunParameters[0].Parameter) {
-        tl.debug('Overriding test run parameters.');
-        const parametersArray = result.RunSettings.TestRunParameters[0].Parameter;
-        parametersArray.forEach(function (parameter) {
-            const key = parameter.$.Name || parameter.$.name;
-            if (overrideParameters[key] && overrideParameters[key].value) {
-                if (parameter.$.Value) {
-                    parameter.$.Value = overrideParameters[key].value;
-                } else {
-                    parameter.$.value = overrideParameters[key].value;
-                }
-            }
-        });
+    var parametersArray;
+    if (result.RunSettings) 
+	{
+		if(result.RunSettings.TestRunParameters && result.RunSettings.TestRunParameters[0] &&
+        result.RunSettings.TestRunParameters[0].Parameter) 
+		{
+			tl.debug('Overriding test run parameters for run settings.');
+			parametersArray = result.RunSettings.TestRunParameters[0].Parameter;
+		}
     }
+	else if(result.TestSettings)
+	{
+	    if(result.TestSettings.Properties && result.TestSettings.Properties[0] &&
+        result.TestSettings.Properties[0].Property) 
+		{
+			tl.debug('Overriding test run parameters for test settings.');
+		    parametersArray = result.TestSettings.Properties[0].Property;
+        }  
+    }	
+    
+    if(parametersArray)
+        {
+            parametersArray.forEach(function (parameter) {
+				const key = parameter.$.Name || parameter.$.name;
+				if (overrideParameters[key] && overrideParameters[key].value) {
+					if (parameter.$.Value) {
+						parameter.$.Value = overrideParameters[key].value;
```

Comment:
Debug log here..

Reply:
added

---
In reply to: [137517119](https://github.com/Microsoft/vsts-tasks/pull/5191#discussion_r137517119) [](ancestors = 137517119)